### PR TITLE
Fixes for ops_kfd

### PR DESCRIPTION
**Pull request by fxkamd**
Two small fixes as I'm getting tinygrad working with KFD.

### CODE REVIEW

**Comment by geohot** [comment]
Linter failing on 150 char line limit, we have a pre-commit hook that should catch things like this. Install with `pre-commit install`

File: tinygrad/runtime/ops_kfd.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 from __future__ import annotations
 from typing import Tuple, Any
+from pathlib import Path
```
Comment (nimlgen) [inline tinygrad/runtime/ops_kfd.py:3]: pathlib is imported already

File: tinygrad/runtime/ops_kfd.py [hunk 2]
```diff
@@ -363,10 +371,12 @@ def _wait_signal(self, signal:hsa.amd_signal_t, timeout=60000):
...
     self.device_id = int(device.split(":")[1]) if ":" in device else 0
-    with open(node_sysfs_path(self.device_id+1, "gpu_id"), "r") as f: self.gpu_id = int(f.read())
-    with open(node_sysfs_path(self.device_id+1, "properties"), "r") as f: self.properties = {line.split()[0]: int(line.split()[1]) for line in f}
+    first_gpu = find_first_gpu_node()
+    with open(node_sysfs_path(self.device_id+first_gpu, "gpu_id"), "r") as f: self.gpu_id = int(f.read())
+    with open(node_sysfs_path(self.device_id+first_gpu, "properties"), "r") as f: self.properties = {line.split()[0]: int(line.split()[1]) for line in f}
     self.drm_fd = os.open(f"/dev/dri/renderD{self.properties['drm_render_minor']}", os.O_RDWR)
-    self.arch = f"gfx{self.properties['gfx_target_version']//100}"
+    target = int(self.properties['gfx_target_version'])
+    self.arch = "gfx%d%x%x" % (target // 10000, (target // 100) % 100, target % 100)
```
Comment (nimlgen) [inline tinygrad/runtime/ops_kfd.py:379]: Should we change that? We get major and minor which are passed to the compiler (like `gfx1100`), why should we care about revision?

File: tinygrad/runtime/ops_kfd.py (hunk 1 above)
Comment (fxkamd) [reply tinygrad/runtime/ops_kfd.py]: Thanks, I'll just use pathlib.Path.

**Comment by fxkamd** [comment]
> Looks good.
> 
> Linter failing on 150 char line limit, we have a pre-commit hook that should catch things like this. Install with `pre-commit install`

I've got most pre-commit tests passing now. I get one failure, not sure if pip installed the wrong version of pytest:

'''strip whitespace.........................................................Passed
ruff.....................................................................Passed
mypy.....................................................................Passed
docs2....................................................................Passed
select GPU tests.........................................................Passed
subset of tests..........................................................Failed
- hook id: tests
- exit code: 4

ERROR: usage: __main__.py [options] [file_or_dir] [file_or_dir] [...]
__main__.py: error: unrecognized arguments: -n=4
  inifile: None
'''

**Comment by geohot** [comment]
`pip -e install .[testing]` should install all you need, I think you are missing `pytest-xdist`

Sorry, we really need to write better docs, it's just in a footnote at the bottom of the readme now.

File: tinygrad/runtime/ops_kfd.py [hunk 3]
```diff
@@ -17,7 +17,14 @@
 libc.munmap.argtypes = [ctypes.c_void_p, ctypes.c_size_t]
 libc.munmap.restype = ctypes.c_int
 
-def node_sysfs_path(node_id, file): return f"/sys/devices/virtual/kfd/kfd/topology/nodes/{node_id}/{file}"
+def is_usable_gpu(gpu_id):
+    try:
```
Comment (geohot) [inline tinygrad/runtime/ops_kfd.py:21]: Hmm, why isn't our linter catching spacing here?

File: tinygrad/runtime/ops_kfd.py [hunk 4]
```diff
@@ -17,7 +17,14 @@
...
+      with gpu_id.open() as f:
+        return int(f.read()) != 0
+    except OSError:
+      return False
+gpus = [g.parent for g in pathlib.Path("/sys/devices/virtual/kfd/kfd/topology/nodes").glob("*/gpu_id") if is_usable_gpu(g)]
```
Comment (geohot) [inline tinygrad/runtime/ops_kfd.py:26]: This should be in the first call to KFDDevice __init__ (with `if KFDDevice.kfd == -1:`) so it doesn't run on import.

**Pull request merged**